drivers/ethernet/dsa_nxp_imx_netc.c

```diff
@@ -0,0 +1,297 @@
+/*
+ * Copyright 2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define LOG_MODULE_NAME dsa_netc
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(LOG_MODULE_NAME, CONFIG_ETHERNET_LOG_LEVEL);
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/kernel.h>
+#include <errno.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/net/ethernet.h>
+#include <zephyr/linker/sections.h>
+#include <zephyr/toolchain/common.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/net/phy.h>
+#include <zephyr/devicetree.h>
+
+#include "fsl_netc_switch.h"
+
+#define DT_DRV_COMPAT nxp_imx_netc_dsa
+#include "dsa_nxp_imx_netc.h"
+
+#define PRV_DATA(ctx) ((struct dsa_netc_data *const)(ctx)->prv_data)
+
+struct dsa_netc_data {
+	int port_num;
+	int port_init_count;
+	int iface_init_count;
+	swt_config_t swt_config;
+	swt_handle_t swt_handle;
+	const struct device *dev_master;
+	netc_cmd_bd_t *cmd_bd;
+};
+
+struct dsa_netc_slave_config {
+	/** MAC address for each LAN{123.,} ports */
+	uint8_t mac_addr[6];
```

Comment:
Will the number of ports vary between IP instances? Or is it fixed at 6?

Reply:
This is to store mac address. But not number.
Thanks.